tests/compare_result.py

```diff
@@ -35,17 +35,24 @@ def error_message(fout_name, suffix, i_iter, abs_err, filename, filename_ref):
 
 @pytest.fixture
 def compare_energyresult(output_dir, rootdir):
-    """Compare dat file output of EnergyResult with the file in reference folder"""
-    def _inner(fout_name, suffix, adpt_num_iter, suffix_ref=None, precision=1E-10):
-        if suffix_ref is None:
-            suffix_ref = suffix
+    """Compare dat file output of EnergyResult with the file in reference folder
+    precision: absolute precision
+    prec_rel: precision relative to average of data_smooth_ref 
+    """
+    def _inner(fout_name, suffix, adpt_num_iter,suffix_ref=None,precision=None,prec_rel=1E-12):
+        if suffix_ref is None :
+            suffix_ref=suffix
         for i_iter in range(adpt_num_iter+1):
             filename     = fout_name + f"-{suffix}_iter-{i_iter:04d}.dat"
             filename_ref = fout_name + f"-{suffix_ref}_iter-{i_iter:04d}.dat"
             path_filename = os.path.join(output_dir, filename)
             E_titles, data_energy, data, data_smooth = read_energyresult_dat(path_filename)
             path_filename_ref = os.path.join(rootdir, 'reference', filename_ref)
             E_titles_ref, data_energy_ref, data_ref, data_smooth_ref = read_energyresult_dat(path_filename_ref)
+            if precision is None:
```

Comment:
Ok, but this should be done only once, in the beginning of the function, not at every iteration i_iter